src/uu/seq/BENCHMARKING.md

```diff
@@ -19,7 +19,57 @@ Finally, you can compare the performance of the two versions of `seq`
 by running, for example,
 
 ```shell
-hyperfine "seq 1000000" "target/release/seq 1000000"
+hyperfine -L seq seq,target/release/seq "{seq} 1000000"
+```
+
+## Interesting test cases
+
+Performance characteristics may vary a lot depending on the parameters,
+and if custom formatting is required. In particular, it does appear
+that the GNU implementation is heavily optimized for positive integer
+outputs (which is probably the most common use case for `seq`).
+
+Specifying a format or fixed width will slow down the
+execution a lot (~15-20 times on GNU `seq`):
+```shell
+hyperfine -L seq seq,target/release/seq "{seq} -f%g 1000000"
+hyperfine -L seq seq,target/release/seq "{seq} -w 1000000"
+```
+
+Floating point increments, or any negative bound, also degrades the
+performance (~10-15 times on GNU `seq`):
+```shell
+hyperfine -L seq seq,./target/release/seq "{seq} 0 0.000001 1"
+hyperfine -L seq seq,./target/release/seq "{seq} -100 1 1000000"
+```
+
+## Optimizations
+
+### Buffering stdout
+
+The original `uutils` implementation of `seq` did unbuffered writes
+to stdout, causing a large number of system calls (and therefore a large amount
+of system time). Simply wrapping `stdout` in a `BufWriter` increased performance
+by about 2 times for a floating point increment test case, leading to similar
+performance compared with GNU `seq`:
+```shell
+taskset -c 0 hyperfine -L seq seq,./seq-main,target/release/seq "{seq} 0 0.1 100000"
```

Comment:
i don't think we need to document the results 
just the commands

Reply:
Dropped.

I saw some results in some other `BENCHMARKING.md` so I just put them there... Might be nice to codify what's expected in those documentation files (maybe in `CONTRIBUTING.md`?). Not a big deal.

Reply:
sure, don't hesitate to make a proposition :)